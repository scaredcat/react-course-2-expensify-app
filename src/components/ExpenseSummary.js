import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';

import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';


export const ExpenseSummary = props => {
  if (props.expensesCount === 0) {
    return null;
  }
  return (
    <div>
      Viewing {props.expensesCount} {props.expensesCount > 1 ? 'expenses' : 'expense'} totalling {numeral(props.expensesTotal / 100).format('$0,0.00')}
    </div>
  );
};

const mapStateToProps = state => {
  const expenses = selectExpenses(state.expenses, state.filters);
  return {
    expensesCount: expenses.length,
    expensesTotal: selectExpensesTotal(expenses)
  };
};

export default connect(mapStateToProps)(ExpenseSummary);
