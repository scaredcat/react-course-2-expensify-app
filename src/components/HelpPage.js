import React from 'react';

const HelpPage = () => (
  <div>
    This is from the Help page component
  </div>
);

export default HelpPage;
