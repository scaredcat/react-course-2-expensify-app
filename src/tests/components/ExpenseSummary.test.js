import React from 'react';
import { shallow } from 'enzyme';

import { ExpenseSummary } from '../../../src/components/ExpenseSummary';

test('it should display summary of a single expense', () => {
  const wrapper = shallow(
    <ExpenseSummary
      expensesCount={1}
      expensesTotal={9434}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

test('it should display summary of more than one expense', () => {
  const wrapper = shallow(
    <ExpenseSummary
      expensesCount={2}
      expensesTotal={9434}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
