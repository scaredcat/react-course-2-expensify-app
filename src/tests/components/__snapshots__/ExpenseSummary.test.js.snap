// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should display summary of a single expense 1`] = `
<div>
  Viewing 
  1
   
  expense
   totalling 
  $94.34
</div>
`;

exports[`it should display summary of more than one expense 1`] = `
<div>
  Viewing 
  2
   
  expenses
   totalling 
  $94.34
</div>
`;
